### ISSUE TITLE
Define variables for an entire notebook

### ISSUE THREAD

**Issue by trevorlangeveldt-novia** [microsoft/azuredatastudio#19903]
When working in a notebook, I am forced to declare the same variable(s) in each code block when I want to apply the same filtering to multiple code blocks. It would be nice to declare "global" variables scoped to a particular notebook so that I can declare and set the variable(s) once and then have it referenced in all code blocks below.

NOTE: I am using the SQL kernel in my notebooks

**Comment by chlafreniere** [comment]
Thanks for the feedback, @trevorlangeveldt-novia. This isn't something that's on our radar at the moment, but we do have the [.NET Interactive extension](https://github.com/dotnet/interactive) in the ADS extension gallery, where you can execute SQL queries and C#/F#/PowerShell code in one notebook (and pass variables between different languages). Feel free to check that out and give us feedback!

**Comment by phierru** [comment]
It should be on your radar though. That's the way Notebooks work. You should be able to store the results of a cell in a variable and then use it later in another cell. You could store the variables in a context. See this approach: https://www.ericgharrison.com/?p=418

**Comment by ishazel** [comment]
Agreed, this ought to work. I got really excited when I discovered parameter cells, but they weren't what I thought.

**Comment by neospud1** [comment]
Anyone know what happened to the ADS version of the extension?  .NET Interactive was renamed to Polyglot Notebooks in VS Code but in ADS, I don't see either.
